src/hats_import/index/run_index.py

```diff
@@ -22,14 +22,11 @@ def run(args, client):
         use_progress_bar=args.progress_bar,
         simple_progress_bar=args.simple_progress_bar,
     ) as step_progress:
-        index_catalog_info = args.to_catalog_info(int(rows_written))
-        write_metadata.write_provenance_info(
-            catalog_base_dir=args.catalog_path,
-            dataset_info=index_catalog_info,
-            tool_args=args.provenance_info(),
-        )
+        index_catalog_info = args.to_table_properties(int(rows_written))
         step_progress.update(1)
-        write_metadata.write_catalog_info(catalog_base_dir=args.catalog_path, dataset_info=index_catalog_info)
+        ## TODO - optionally write out arguments file
+        step_progress.update(1)
+        index_catalog_info.to_properties_file(args.catalog_path)
```

Comment:
Should we move this call to line 26, right after `to_table_properties`, for consistency with the other pipelines? We could then remove this extra `step_progress.update` so that we only have 4 steps, as defined in the `print_progress` arguments.